lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
```

Comment:
I don't think it's worth it to make the code that complicated when we just checking if it's an array (but alas, such is JavaScript). Plus, that would be adding a third layer of slowness if `values` is a plain object.